embulk-output-redshift/src/main/java/org/embulk/output/redshift/RedshiftCopyBatchInsert.java

```diff
@@ -82,7 +83,9 @@ public RedshiftCopyBatchInsert(JdbcOutputConnector connector,
         this.credentialsProvider = credentialsProvider;
         this.s3 = new AmazonS3Client(credentialsProvider);  // TODO options
         this.sts = new AWSSecurityTokenServiceClient(credentialsProvider);  // options
-        this.executorService = Executors.newCachedThreadPool();
+        this.executorService = maxS3UploadThreadsPerTask != null
+                ? Executors.newFixedThreadPool(maxS3UploadThreadsPerTask)
+                : Executors.newCachedThreadPool();
         this.uploadAndCopyFutures = new ArrayList<Future<Void>>();
```

Comment:
The original PR change thread pool from `newThreadPool` to `Executors.newFixedThreadPool(threadMaximum)`. 
Why you change this part from the origianl?

Reply:
For the same reason as above, if `max_s3_upload_threads_per_task` is not specified, continue to use `newCachedThreadPool` to keep the original behavior.

Reply:
OK. Thanks